lib/neptune/incremental_learning/incremental_learning.py

```diff
@@ -0,0 +1,174 @@
+    :param test_data:
+    :param class_names:
+    :param input_shape: the input shape of model
+    """
+    il_config = IncrementalConfig()
+
+    results = []
+    for model_url in il_config.model_urls.split(';'):
+        precision, recall, all_precisions, all_recalls = model(
+            model_path=model_url,
+            test_dataset=test_data,
+            class_names=class_names,
+            input_shape=input_shape)
+
+        result = {
+            "format": "pb",
+            "url": remove_path_prefix(model_url, il_config.data_path_prefix),
+            "metrics": {
+                "recall": recall,
+                "precision": precision
+            }
+        }
+        results.append(result)
+
+    message = {
+        "name": il_config.worker_name,
+        "namespace": il_config.namespace,
+        "ownerName": il_config.job_name,
+        "ownerKind": K8sResourceKind.INCREMENTAL_JOB.value,
+        "kind": "eval",
+        "status": K8sResourceKindStatus.COMPLETED.value,
+        "results": results
+    }
+
+    LCClient.send(il_config.worker_name, message)
+
+
+class TSModel(TSLittleModel):
+    def __init__(self, preprocess=None, postprocess=None, input_shape=(0, 0),
+                 create_input_feed=None, create_output_fetch=None):
+        TSLittleModel.__init__(self, preprocess, postprocess, input_shape,
+                               create_input_feed, create_output_fetch)
+
+
+class InferenceResult:
+    def __init__(self, is_hard_example, infer_result):
+        self.is_hard_example = is_hard_example
+        self.infer_result = infer_result
+
+
+class Inference:
+    def __init__(self, model: TSModel, hard_example_mining_algorithm=None):
+        if hard_example_mining_algorithm is None:
+            hem_name = BaseConfig.hem_name
+
+            if hem_name == "IBT":
+                threshold_box = float(neptune.context.get_hem_parameters(
+                    "threshold_box", 0.8
+                ))
+                threshold_img = float(neptune.context.get_hem_parameters(
+                    "threshold_img", 0.8
+                ))
+                hard_example_mining_algorithm = IBTFilter(threshold_img,
+                                                          threshold_box)
+            elif hem_name == "CrossEntropy":
+                threshold_cross_entropy = float(
+                    neptune.context.get_hem_parameters(
+                        "threshold_cross_entropy", 0.5
+                    )
+                )
+                hard_example_mining_algorithm = CrossEntropyFilter(
+                    threshold_cross_entropy)
+            else:
+                hard_example_mining_algorithm = ThresholdFilter()
+        self.hard_example_mining_algorithm = hard_example_mining_algorithm
+        self.model = model
+
+    def inference(self, img_data) -> InferenceResult:
+        result = self.model.inference(img_data)
+        bboxes = deal_infer_rsl(result)
```

Comment:
bboxes only in image detection, other infer types no bbox

Reply:
had modify